presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQueryFactory.java

```diff
@@ -92,14 +94,16 @@ public DispatchQuery createDispatchQuery(
             String query,
             PreparedQuery preparedQuery,
             Slug slug,
-            ResourceGroupId resourceGroup)
+            ResourceGroupId resourceGroup,
+            Tracer tracer)
     {
         WarningCollector warningCollector = warningCollectorFactory.create();
+        URI queryLocation = locationFactory.createQueryLocation(session.getQueryId());
```

Comment:
discussed offline, looks like using NodeInfo might be a better choice to get host info